tilings/strategies/sliding.py

```diff
@@ -0,0 +1,276 @@
+from typing import Callable, DefaultDict, Dict, Iterator, List, Optional, Tuple
+
+from comb_spec_searcher.strategies.strategy import (
+    DisjointUnionStrategy,
+    StrategyFactory,
+)
+from tilings import GriddedPerm, Tiling
+from tilings.algorithms import (
+    get_col_info,
+    gp_slide,
+    gp_slide_inverse,
+    slidable_pairs,
+    slide_assumption,
+    slide_column,
+)
+
+
+def _tiling_identity_map(tiling: Tiling) -> Tiling:
+    return tiling
+
+
+def _tiling_reverse(tiling: Tiling) -> Tiling:
+    t: Tiling = tiling.reverse()
+    return t
+
+
+def _tiling_rotate90(tiling: Tiling) -> Tiling:
+    t: Tiling = tiling.rotate90()
+    return t
+
+
+def _tiling_rotate90_inverse(tiling: Tiling) -> Tiling:
+    t: Tiling = tiling.rotate270()
+    return t
+
+
+def _tiling_rotate90_and_reverse(tiling: Tiling) -> Tiling:
+    t: Tiling = tiling.rotate90().reverse()
+    return t
+
+
+def _tiling_rotate90_and_reverse_inverse(tiling: Tiling) -> Tiling:
+    t: Tiling = tiling.reverse().rotate270()
+    return t
+
+
+def _gp_identity_map(gp: GriddedPerm) -> GriddedPerm:
+    return gp
+
+
+def _gp_reverse(c: int) -> Callable[[GriddedPerm], GriddedPerm]:
+    def _tmp_func(gp: GriddedPerm) -> GriddedPerm:
+        return GriddedPerm(gp.patt.reverse(), ((c - x - 1, 0) for x, _ in gp.pos))
+
+    return _tmp_func
+
+
+def _gp_rotate90(r: int) -> Callable[[GriddedPerm], GriddedPerm]:
+    def _tmp_func(gp: GriddedPerm) -> GriddedPerm:
+        return GriddedPerm(gp.patt.rotate(), ((y, 0) for _, y in gp.pos))
+
+    return _tmp_func
+
+
+def _gp_rotate90_inverse(r: int) -> Callable[[GriddedPerm], GriddedPerm]:
+    def _tmp_func(gp: GriddedPerm) -> GriddedPerm:
+        return GriddedPerm(gp.patt.rotate(-1), ((0, x) for x, _ in gp.pos))
+
+    return _tmp_func
+
+
+def _gp_rotate90_and_reverse(r: int) -> Callable[[GriddedPerm], GriddedPerm]:
+    def _tmp_func(gp: GriddedPerm) -> GriddedPerm:
+        return GriddedPerm(
+            gp.patt.rotate(1).reverse(), ((r - y - 1, 0) for _, y in gp.pos)
+        )
+
+    return _tmp_func
+
+
+def _gp_rotate90_and_reverse_inverse(r: int) -> Callable[[GriddedPerm], GriddedPerm]:
+    def _tmp_func(gp: GriddedPerm) -> GriddedPerm:
+        return GriddedPerm(
+            gp.patt.reverse().rotate(-1), ((0, r - x - 1) for x, _ in gp.pos)
+        )
+
+    return _tmp_func
```

Comment:
I understand why you have given these their own methods. The methods on `GriddedPerm` maybe should take into account the context in which the symmetry is taken in a simple way. This is probably an issue for a future PR.